mod-record-specifications-validator/src/main/java/org/folio/rspec/validation/validator/marc/impl/NonRepeatableSubfieldRuleValidator.java

```diff
@@ -0,0 +1,67 @@
+package org.folio.rspec.validation.validator.marc.impl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.folio.rspec.domain.dto.DefinitionType;
+import org.folio.rspec.domain.dto.SeverityType;
+import org.folio.rspec.domain.dto.SpecificationFieldDto;
+import org.folio.rspec.domain.dto.SubfieldDto;
+import org.folio.rspec.domain.dto.ValidationError;
+import org.folio.rspec.i18n.TranslationProvider;
+import org.folio.rspec.utils.SpecificationUtils;
+import org.folio.rspec.validation.validator.SpecificationRuleCode;
+import org.folio.rspec.validation.validator.SpecificationRuleValidator;
+import org.folio.rspec.validation.validator.marc.model.MarcRuleCode;
+import org.folio.rspec.validation.validator.marc.model.MarcSubfield;
+
+public class NonRepeatableSubfieldRuleValidator
+  implements SpecificationRuleValidator<List<MarcSubfield>, SpecificationFieldDto> {
+
+  private static final String CODE_KEY = "code";
+
+  private final TranslationProvider translationProvider;
+
+  NonRepeatableSubfieldRuleValidator(TranslationProvider translationProvider) {
+    this.translationProvider = translationProvider;
+  }
+
+  @Override
+  public List<ValidationError> validate(List<MarcSubfield> marcSubfields, SpecificationFieldDto specification) {
+    var nonRepeatableSubfields = SpecificationUtils.nonRepeatableSubfields(specification.getSubfields());
+
+    return marcSubfields.stream()
+      .collect(Collectors.groupingBy(MarcSubfield::code))
```

Comment:
This grouping was already done on record conversion.
So. you can use `marcSubfield.reference().getSubfieldIndex() > 0` as a condition to identify if some subfield is appeared more then once.

Reply:
done